rpcserver.go

```diff
@@ -1908,6 +1908,33 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 			err)
 	}
 
+	var channelType *lnwire.ChannelType
+	switch in.CommitmentType {
+	case lnrpc.CommitmentType_UNKNOWN_COMMITMENT_TYPE:
+		break
+
+	case lnrpc.CommitmentType_LEGACY:
+		channelType = new(lnwire.ChannelType)
```

Comment:
Any reason you went w/ this syntax vs just declaring the value above and passing in the pointer below? Would shave off a few lines.

Reply:
So that the pointer is still nil when an explicit channel type isn't provided and explicit channel negotiation isn't attempted.